docs/use-cases.rst

```diff
@@ -0,0 +1,668 @@
+Use Cases
+=========
+
+.. include:: _include.rst
+
+The following use cases represent the kinds of interaction users may have with
+Truffle and other tools. These use cases serve to help validate proposals for
+changes to Truffle's artifact format.
+
+These use cases are written primarily for Truffle, as the primary producer
+and consumer of artifacts, but should apply generally for any and all
+tooling for smart contract / decentralized application development.
+
+Since the process to develop and deploy a decentralized applications can
+include many participants, these use cases refer to several different user
+profiles:
+
+- Smart Contract Developer
+- Frontend Developer
+- Business Stakeholder
+- Auditor
+
+Not all of these users interact with development tooling directly, or in the
+same ways. As such, these use cases should specify appropriately.
+
+This list may not be exhaustive, but should cover the vast majority of required
+behaviors.
+
+
+Saving artifacts
+----------------
+
+Users and tooling must be able to save new and update existing |Artifact|
+files.
+
+As a core behavior, tooling must facilitate the internal record-keeping of all
+known |ContractTypes| and |ContractInstances|, updating them appropriately
+when they change, or when knowledge of them changes.
+
+Beyond that, developers may also need to manually generate new artifacts for
+previously unknown contracts.
+
+Since a smart contract blockchain network is effectively a distributed global
+database, developers may write their applications for the express or implicit
+purpose of accessing other contract instances on that network.
+
+Developers should be able to leverage tooling to interface with these external
+contracts, either common, public libraries, or contract instances for other
+applications. Tooling should integrate well with code written outside a given
```

Comment:
What do you think about  modifying the break at `. . .contracts, either . . ` to be `...contracts, which may be...`